pinot-core/src/main/java/org/apache/pinot/core/util/GenericRowSerDeUtils.java

```diff
@@ -0,0 +1,326 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.util;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.pinot.segment.local.segment.memory.PinotDataBuffer;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * Utility methods for serde of {@link GenericRow}
+ * Deserialization assumes it is deserializing from a {@link PinotDataBuffer}
+ */
+public final class GenericRowSerDeUtils {
+
+  private GenericRowSerDeUtils() {
+
+  }
+
+  /**
+   * Serialize the given GenericRow
+   * @param genericRow GenericRow to serialize
+   * @param fieldSpecs the fields to serialize
+   * @return serialized bytes
+   */
+  public static byte[] serializeGenericRow(GenericRow genericRow, List<FieldSpec> fieldSpecs) {
```

Comment:
Suggest merging the length calculation and the serialize so that we only need one switch case, and no need to do UTF-8 encoding twice. Don't see much value breaking them into 2 passes

Reply:
i can store the values in an intermediate structure, and save one UTF-8 encoding. But how would i save on the switch? I thought I need to do the length calculation first, so that I can create the ByteBuffer using numBytes
```
byte[] genericRowBytes = new byte[numBytes];
    ByteBuffer byteBuffer = ByteBuffer.wrap(genericRowBytes).order(PinotDataBuffer.NATIVE_ORDER);
```

Reply:
You are right, I didn't realize the length is for the entire row. Let's keep it 2 passes then